src/test/java/de/tum/in/www1/artemis/metis/MessageIntegrationTest.java

```diff
@@ -177,23 +180,33 @@ void testCreateConversationPost(CourseInformationSharingConfiguration courseInfo
         verify(websocketMessagingService, timeout(2000).times(2)).sendMessageToUser(anyString(), anyString(), any(PostDTO.class));
     }
 
-    @Test
-    @WithMockUser(username = TEST_PREFIX + "student1", roles = "USER")
-    void testCreateConversationPostInCourseWideChannel() throws Exception {
-        Channel channel = conversationUtilService.createCourseWideChannel(course, "test");
-        conversationUtilService.addParticipantToConversation(channel, TEST_PREFIX + "student2");
+    @ParameterizedTest
+    @ValueSource(booleans = { false, true })
+    @WithMockUser(username = TEST_PREFIX + "instructor1", roles = "INSTRUCTOR")
+    void testCreateConversationPostInCourseWideChannel(boolean isAnnouncement) throws Exception {
+        Channel channel = conversationUtilService.createCourseWideChannel(course, "test", isAnnouncement);
+        ConversationParticipant otherParticipant = conversationUtilService.addParticipantToConversation(channel, TEST_PREFIX + "student2");
         ConversationParticipant author = conversationUtilService.addParticipantToConversation(channel, TEST_PREFIX + "student1");
 
         Post postToSave = new Post();
         postToSave.setAuthor(author.getUser());
         postToSave.setConversation(channel);
+        postToSave.setContent("message");
 
         Post createdPost = request.postWithResponseBody("/api/courses/" + courseId + "/messages", postToSave, Post.class, HttpStatus.CREATED);
         checkCreatedMessagePost(postToSave, createdPost);
 
         // conversation participants should be notified via one broadcast
         verify(websocketMessagingService, never()).sendMessageToUser(anyString(), anyString(), any(PostDTO.class));
         verify(websocketMessagingService, timeout(2000).times(1)).sendMessage(eq("/topic/metis/courses/" + courseId), any(PostDTO.class));
+
+        if (isAnnouncement) {
+            verify(mailService, timeout(2000).times(1)).sendNotification(any(ConversationNotification.class), eq(otherParticipant.getUser()), any(Post.class));
+            verify(mailService, timeout(2000).times(1)).sendNotification(any(ConversationNotification.class), eq(author.getUser()), any(Post.class));
```

Comment:
Should the author really receive an email?
I always hate these notifications that notify me about an action I've done.

Reply:
Yes, the author should receive an email. However, I added a check for the email settings of the author, so they only get notified if they have email notifications enabled for the notification type.